packages/dma-contracts/contracts/actions/common/PullToken.sol

```diff
@@ -20,7 +20,12 @@ contract PullToken is Executable {
   function execute(bytes calldata data, uint8[] memory) external payable override {
     PullTokenData memory pull = parseInputs(data);
 
-    IERC20(pull.asset).safeTransferFrom(pull.from, address(this), pull.amount);
+    IERC20 token = IERC20(pull.asset);
+
+    if (pull.amount == type(uint256).max) {
+      pull.amount = token.balanceOf(pull.from);
```

Comment:
One of the use cases here is that frontend would set up allowance before the tx. And thus exact amount is not known before the tx (it could be aToken for example) the allowance is slightly bigger to cover any additional amount. But if we would pull this allowed amount it might turn out there is little less than the allowance. So I would stick to the token balance, as it's specific to use cases we have right now.

Reply:
Thanks for the insight!